src/extension.ts

```diff
@@ -323,16 +323,18 @@ export async function activate(context: vscode.ExtensionContext) {
 
         wakePort = await getPort();
 
-        let wakePath: string = pathToExecutable ?? "wake";
-        if (!pathToExecutable && cwd !== undefined) {
-            wakePath = path.join(cwd, "wake");
+        if (pathToExecutable !== null) {
+            cwd = path.dirname(pathToExecutable);
         }
 
+        const wakePath: string = cwd ? path.join(cwd, "wake") : "wake";
+
         outputChannel.appendLine(`Running '${wakePath} lsp --port ${wakePort}' (${wakeVersion})`);
         if (cwd === undefined) {
-            wakeProcess = spawn(wakePath, ["lsp", "--port", String(wakePort),], { shell: true, stdio: ['ignore', 'ignore', 'pipe'], env: { ...process.env, PYTHONIOENCODING: 'utf8' } });
+            wakeProcess = execa('wake', ["lsp", "--port", String(wakePort)], { shell: true, stdio: ['ignore', 'ignore', 'pipe'], env: { ...process.env, PYTHONIOENCODING: 'utf8' } });
         } else {
-            wakeProcess = spawn(wakePath, ["lsp", "--port", String(wakePort)], { cwd, shell: true, stdio: ['ignore', 'ignore', 'pipe'], env: { ...process.env, PYTHONIOENCODING: 'utf8' }});
+            const cmd = process.platform === "win32" ? ".\\wake" : "./wake";
```

Comment:
On Unix it's needed to launch with `./wake` to run the local executable and not global. Windows uses a different format.